modules/users/server/controllers/users/users.authentication.server.controller.js

```diff
@@ -20,11 +20,22 @@ var noReturnUrls = [
  * Signin after passport authentication
  */
 exports.signin = function (req, res, next) {
-  passport.authenticate('local', function (err, user, info) {
+  passport.authenticate('local', async function (err, user, info) {
     if (!user) {
       user = new User(req.body);
       user.provider = 'local';
-      user.save();
+      try {
+        await user.save();
```

Comment:
This is the main good fix - make sure we care about the result of saving the user.